crates/ruma-common/src/api/error.rs

```diff
@@ -294,6 +298,42 @@ pub enum HeaderDeserializationError {
         /// The value we instead received and rejected.
         unexpected: String,
     },
+
+    /// The `Content-Type` header for a `multipart/mixed` response is missing the `boundary`
+    /// attribute.
+    #[error(
+        "The `Content-Type` header for a `multipart/mixed` response is missing the `boundary` attribute"
+    )]
+    MissingMultipartBoundary,
+}
+
+/// An error when deserializing a `multipart/mixed` response.
+#[derive(Debug, Error)]
+#[non_exhaustive]
+pub enum MultipartMixedDeserializationError {
+    /// There were not the number of body parts that were expected.
+    #[error(
+        "multipart/mixed response does not have enough body parts, \
+         expected {expected}, found {found}"
```

Comment:
In case we do decide to enforce that there should only be two parts:
```suggestion
        "multipart/mixed response does not have the correct  \
         number of body parts, expected {expected}, found {found}"
```